### ISSUE TITLE
Activation Code is blank

### ISSUE THREAD

**Issue by ahaghgoo** [wftllc/hahastream#8]
I was able to compile and side load this on my AppleTV, but upon launching the HeHe app on my AppleTV, I hit the activation screen with a blank Activation Code box.  How do I get it to show a code that then I can activate?

**Comment by lavenj** [comment]
Can you copy and paste any log output from the xcode debug area? View menu => Debug Area => Activate Console

**Comment by ahaghgoo** [comment]
Yeah, it’s a repeated request for activation: 

pollForRegistrationKey()
pollForActivation(deviceKey:)
api error Device not activated. Please activate it at http://hehestreams.com/activate
pollForActivation(deviceKey:)
api error Device not activated. Please activate it at http://hehestreams.com/activate
pollForActivation(deviceKey:)
api error Device not activated. Please activate it at http://hehestreams.com/activate
pollForActivation(deviceKey:)
api error Device not activated. Please activate it at http://hehestreams.com/activate
pollForActivation(deviceKey:)
api error Device not activated. Please activate it at http://hehestreams.com/activate
pollForActivation(deviceKey:)
api error Device not activated. Please activate it at http://hehestreams.com/activate
pollForActivation(deviceKey:)
api error Device not activated. Please activate it at http://hehestreams.com/activate
pollForActivation(deviceKey:)
api error Device not activated. Please activate it at http://hehestreams.com/activate
pollForActivation(deviceKey:)
api error Device not activated. Please activate it at http://hehestreams.com/activate
pollForActivation(deviceKey:)
api error Device not activated. Please activate it at http://hehestreams.com/activate
pollForActivation(deviceKey:)
api error Device not activated. Please activate it at http://hehestreams.com/activate

--
Afshin Haghgoo
ahaghgoo@ucla.edu

**Comment by ahaghgoo** [comment]
When I simulated it within XCode, it gave me a code which I activated.  Not sure if that messed anything up when I then side loaded it up on my AppleTV?  

Thanks again for the consideration!  


--
Afshin Haghgoo
ahaghgoo@ucla.edu

**Comment by lavenj** [comment]
Can you build again with the latest code and let me know what happens/post the console logs again?

Thanks!

**Comment by ahaghgoo** [comment]
Unfortunately, I am not getting an activation code when running on my AppleTV.  

When I simulate the build on my laptop, I get an activation code.  Not sure why there’s a discrepancy.  

The console is running the same messages.  As far as errors go, I only got 3 during the build time: 

~/Documents/GitHub/hahastream/Haha Stream/HahaService/Models/HahaError.swift:9:13: Value ‘dict' was defined but never used; consider replacing with boolean test
~/Documents/GitHub/hahastream/Haha Stream/Features/Login/LoginInteractor.swift:26:13: Value ‘ident' was defined but never used; consider replacing with boolean test
~/Documents/GitHub/hahastream/Haha Stream/Extensions/Game+LocalImage.swift:12:8: Variable ‘__FIXME_REIMPLEMENT_THIS' was never used; consider replacing with '_' or removing it

Not sure if any of these help.  I sincerely appreciate your consideration on this!!  

-A

--
Afshin Haghgoo
ahaghgoo@ucla.edu

**Comment by lavenj** [comment]
Yeah, can you please download the latest version of the code and install again?

Then send me the console output (View menu => Debug Area => Activate Console) same as first time

Thanks!

**Comment by ahaghgoo** [comment]
Downloaded latest version; same results: 

pollForRegistrationKey(); identifier: CC6F1C4D-78FB-4D21-B6FE-02D0D8088C9D
pollForActivation(deviceKey:)
api error Device not activated. Please activate it at http://hehestreams.com/activate
pollForActivation(deviceKey:)
api error Device not activated. Please activate it at http://hehestreams.com/activate
pollForActivation(deviceKey:)
api error Device not activated. Please activate it at http://hehestreams.com/activate
pollForActivation(deviceKey:)
api error Device not activated. Please activate it at http://hehestreams.com/activate
pollForActivation(deviceKey:)
api error Device not activated. Please activate it at http://hehestreams.com/activate
pollForActivation(deviceKey:)
api error Device not activated. Please activate it at http://hehestreams.com/activate
pollForActivation(deviceKey:)

--
Afshin Haghgoo
ahaghgoo@ucla.edu

**Comment by pasta15892** [comment]
Same issue here.

```
pollForRegistrationKey(); identifier: [REMOVED]
pollForActivation(deviceKey:)
api error Device not activated. Please activate it at http://hehestreams.com/activate
pollForActivation(deviceKey:)
api error Device not activated. Please activate it at http://hehestreams.com/activate
pollForActivation(deviceKey:)
...
```

**Comment by lavenj** [comment]
Weird. So basically there's no error message AND there's no activation key? Perfect!

I have pushed an update that adds a little more logging. If you can download the latest code and send me the output again I'll take a look.

@pasta15892 might end up helping if you could include the identifier -- you can reset it later by uninstalling and reinstalling the app.

**Comment by ahaghgoo** [comment]
Here’s what I get: 

pollForRegistrationKey(); identifier: C52A3714-C498-4184-B757-D8E5D459779B
Headers: ["X-App": "TvOS App (PacMan TV)", "X-Device-Name": "Apple TV", "X-Device-System": "tvOS", "X-Device-Version": "11.2.1", "X-App-Version": "1b1", "X-Device-CPU": "ARM/x86/x64", "X-Id": "C52A3714-C498-4184-B757-D8E5D459779B-b"]
getDeviceRegistrationKey() success: Optional("8NZM9F; [\"key\": 8NZM9F]")
pollForActivation(deviceKey:) 8NZM9F; ["key": 8NZM9F]
Headers: ["X-App": "TvOS App (PacMan TV)", "X-Device-Name": "Apple TV", "X-Device-System": "tvOS", "X-Device-Version": "11.2.1", "X-App-Version": "1b1", "X-Device-CPU": "ARM/x86/x64", "X-Id": "C52A3714-C498-4184-B757-D8E5D459779B-b"]
api error Device not activated. Please activate it at http://hehestreams.com/activate
pollForActivation(deviceKey:) 8NZM9F; ["key": 8NZM9F]
Headers: ["X-App": "TvOS App (PacMan TV)", "X-Device-Name": "Apple TV", "X-Device-System": "tvOS", "X-Device-Version": "11.2.1", "X-App-Version": "1b1", "X-Device-CPU": "ARM/x86/x64", "X-Id": "C52A3714-C498-4184-B757-D8E5D459779B-b"]
api error Device not activated. Please activate it at http://hehestreams.com/activate

--
Afshin Haghgoo
ahaghgoo@ucla.edu

**Comment by ahaghgoo** [comment]
I just took the key in the debug log and activated the device with my account online and now it’s working (mini success!!).  But the deactivation menu item on the tvOS app doesn’t work properly and I had to delete the app and reinstall it in order to return to the activation screen with the blank code box.  But I was able to reactivate it with the code displayed in the log for the second install of the current version.  
 
--
Afshin Haghgoo
ahaghgoo@ucla.edu

**Comment by lavenj** [comment]
@ahaghgoo glad you got it working, at least.

I did get the bug: If the device was in dark interface mode (ie at night time or set that way manually), it was displaying the activation key in white on a white background. Hilarious. Fixed in above commit.

The deactivation issue is on the hehe end, but since it doesn't seem to affect anything I haven't bothered with it.

@ahaghgoo @pasta15892 thanks for the help
